File: 2089-maximum-matrix-sum/2089-maximum-matrix-sum.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        int negativecount=0;
        long sum=0;
        long minAbs=Long.MAX_VALUE;
        for(int[] row:matrix){
            for(int i:row){
                sum+=Math.abs(i);
                if(i<0)negativecount++;
                minAbs=Math.min(minAbs,Math.abs(i));
            }
        }
        if(negativecount%2==1)sum-=2*minAbs;
        return sum;
    }
}

File: 2089-maximum-matrix-sum/2089-maximum-matrix-sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        totalsum=0
        negativecount=0
        minAbs=float('inf')
        for row in matrix:
                for i in row:
                    totalsum+=abs(i)
                    if i<0:
                        negativecount+=1 
                    minAbs=min(minAbs,abs(i))
        print(totalsum,minAbs)
        if negativecount%2!=0:
            totalsum-=2*minAbs 
        print(totalsum)
        return totalsum    
       